### PR TITLE
mpsl: mpsl_fem: Remove direct SPI pin control

### PR DESCRIPTION
**Pull request by piotrslawecki**
Removed direct SPI pin control as it is being handled
by the SPI driver and PINCTRL.

Signed-off-by: Piotr Sławęcki <piotr.slawecki@nordicsemi.no>

### CODE REVIEW

File: subsys/mpsl/fem/mpsl_fem.c [hunk 1]
```diff
@@ -311,41 +311,7 @@ static int fem_nrf21540_gpio_configure(void)
 #endif
 
 #if DT_NODE_HAS_STATUS(MPSL_FEM_SPI_IF, okay)
```
Comment (jciupis) [inline subsys/mpsl/fem/mpsl_fem.c]: This `#if` block can be removed altogether.

Comment (piotrslawecki) [reply subsys/mpsl/fem/mpsl_fem.c]: My reasoning behind leaving it was that when the support for spi control comes, something might need to be done about the spi interface, which would have it's place there. If it's unnecessary and messy, I'll remove the entire block

Comment (jciupis) [reply subsys/mpsl/fem/mpsl_fem.c]: When the day comes (provided that it actually does come) and this line is needed, we'll add it again. No need to keep it now :)

Comment (piotrslawecki) [reply subsys/mpsl/fem/mpsl_fem.c]: Alright, removed the entire block :)

**Pull request merged**